os_stub/cryptlib_openssl/pk/x509.c

```diff
@@ -2268,6 +2267,20 @@ bool libspdm_set_subject_name(X509_NAME *x509_name, char *subject_name)
     return true;
 }
 
+/*
+ * X509_REQ_INFO is not a publicly exposed part of OpenSSL
+ * In order to allow reading the subject we copy part of the struct here
+ *
+ * TODO: Stop accessing private parts of OpenSSL
+ */
+typedef struct X509_req_info_st {
+    ASN1_ENCODING enc;          /* cached encoding of signed part */
+    ASN1_INTEGER *version;      /* version, defaults to v1(0) so can be NULL */
+    X509_NAME *subject;         /* certificate request DN */
+    X509_PUBKEY *pubkey;        /* public key of request */
+    /* ... */
+} libspdm_X509_req_info_st;
```

Comment:
I dont think it is a good idea to copy the data structure from openssl.

What if the internal structure is changed in new version?

Reply:
In general, if there is error, I hope it can be detected at build time (compilation error), but not runtime (crash).

IMHO, copying data structure brings bigger risk due to runtime error.